Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -2202,11 +2211,19 @@ void FunctionDefinitionWriter::visit(AST::PhonyAssignmentStatement& statement)
 
 void FunctionDefinitionWriter::visit(AST::ReturnStatement& statement)
 {
-    m_stringBuilder.append("return");
+    if (m_cachedEntryPointStage == ShaderStage::Fragment && m_callGraph.ast().usesFragDepth())
+        m_stringBuilder.append(m_cachedFragmentOutputName, " __wgslFragmentReturnResult = ");
+    else
+        m_stringBuilder.append("return");
     if (statement.maybeExpression()) {
         m_stringBuilder.append(" ");
         visit(*statement.maybeExpression());
     }
+
+    if (m_cachedEntryPointStage == ShaderStage::Fragment && m_callGraph.ast().usesFragDepth()) {
```

Comment:
Same as above, maybe there is a much better way to achieve this, I defer to @tadeuzagallo